benchmark/csharp/Tpch/VectorFunctions.cs

```diff
@@ -3,49 +3,58 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Diagnostics;
 using Apache.Arrow;
+using Microsoft.Data;
 
 namespace Tpch
 {
     internal static class VectorFunctions
     {
-        internal static DoubleArray ComputeTotal(DoubleArray price, DoubleArray discount, DoubleArray tax)
+        internal static PrimitiveColumn<double> ComputeTotal(PrimitiveColumn<double> price, PrimitiveColumn<double> discount, PrimitiveColumn<double> tax)
         {
             if ((price.Length != discount.Length) || (price.Length != tax.Length))
             {
                 throw new ArgumentException("Arrays need to be the same length");
             }
 
-            int length = price.Length;
-            var builder = new DoubleArray.Builder().Reserve(length);
-            ReadOnlySpan<double> prices = price.Values;
-            ReadOnlySpan<double> discounts = discount.Values;
-            ReadOnlySpan<double> taxes = tax.Values;
-            for (int i = 0; i < length; ++i)
+            PrimitiveColumn<double> ret = new PrimitiveColumn<double>("Prices", price.Length);
+
+            foreach (ReadOnlyMemory<double> readOnlyTax in tax.GetReadOnlyDataBuffers())
```

Comment:
These nested foreach statements aren't really correct, are they? If there was more than 1 buffer, you wouldn't get the correct answer.

Reply:
Yup, this is wrong. Part of why I marked it WIP :). CI passes now, so I'm cleaning it all up.